src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,210 @@
+ *           last position. This is the error here.
+ *       </li>
+ *       <li>
+ *           Key and subkey are in correct order here, because only keys are relevant.
+ *           Therefore on line 5 you have only "key" an nothing behind.
+ *           On line 6 you have "key." The dot is on position 46 which is higher than nothing.
+ *       </li>
+ *   </ul>
+ */
+@StatelessCheck
+public class OrderedPropertiesCheck extends AbstractFileSetCheck {
+
+    /**
+     * Localization key for check violation.
+     */
+    public static final String MSG_KEY = "properties.notSorted.property";
+    /**
+     * Localization key for IO exception occurred on file open.
+     */
+    public static final String UNABLE_TO_OPEN_FILE = "Unable to open file: ";
+    /**
+     * Pattern matching single space.
+     */
+    private static final Pattern SPACE_PATTERN = Pattern.compile(" ");
+
+    /**
+     * Construct the check with default values.
+     */
+    public OrderedPropertiesCheck() {
+        setFileExtensions("properties");
+    }
+
+    /**
+     * Processes the file and check order.
+     * @param file the file to be processed
+     * @param fileText the contents of the file.
+     * @throws CheckstyleException
+     * I like to keep the order - keySet is not possible
+     * @noinspection EnumerationCanBeIteration
+     */
+    @Override
+    protected void processFiltered(File file, FileText fileText) throws CheckstyleException {
+        final SequencedProperties properties = new SequencedProperties();
+        try (InputStream inputStream = Files.newInputStream(file.toPath())) {
+            properties.load(inputStream);
+        }
+        catch (IOException ex) {
+            throw new CheckstyleException(UNABLE_TO_OPEN_FILE + file.getPath(), ex);
+        }
+
+        String previousProp = "";
+
+        final Enumeration<Object> keys = properties.keys();
+
+        while (keys.hasMoreElements()) {
+
+            final String propKey = (String) keys.nextElement();
+
+            //if (previousProp.compareTo(propKey) > 0) {
+            if (String.CASE_INSENSITIVE_ORDER.compare(previousProp, propKey) > 0) {
+
+                final int lineNumber = getLineNumber(fileText, propKey);
+                log(lineNumber, MSG_KEY, propKey, previousProp);
+            }
+
+            previousProp = propKey;
+        }
+    }
+
+    /**
+     * Method returns the line number where the key is detected.
+     *
+     * @param fileText {@link FileText} object contains the lines to process
+     * @param keyName  key name to look for
+     * @return line number of first occurrence. If no key found in properties file, 0 is returned
+     */
+    private static int getLineNumber(FileText fileText, String keyName) {
+        final Pattern keyPattern = getKeyPattern(keyName);
+        int humanIndex = 0;
+
+        for (int index = 0; index < fileText.size(); index++) {
```

Comment:
Question was never answered at https://github.com/checkstyle/checkstyle/pull/6343#discussion_r245650215.
> How reliable is this at finding the line number?
What if my key was `a` and I had a bunch of values before `a=` that had `a` in it? Is it possible it won't pick the right line?
Can keys be multi-line like values?
If the same key is used twice, it will report the wrong line.
If this isn't reliable, we shouldn't use it as people will just complain of false line numbers.

Reply:
I was looking into  UniquePropertiesCheck. I found also 2 other implementation having the same pattern.
This implementation relies on the FileText. It loops and count what this implementation returns. My implementation will stop if the first match was found. (no multi key)
I can try to implement one more test to prove your concern above. I will try to verify the getKeyPattern() method. Because this seems unclear here.

Reply:
I implemented for getLineNumber two new validations.
1) check previous property key followed by property key
2) the starting line number is the line number of the last found problem.
In testfile InputOrderedProperties2EmptyValue.properties I inserted repeating keys to verify the implementation.
This should help to find the right line number for reporting a problem.

Reply:
What about this concern?
> Can keys be multi-line like values?

Reply:
Today I worked on multi-line key values.
Java Property class can handle this. The key string contains a real newline. The FileText returns an already quoted string `\\n`. Therefore the regexp will not find a match. I solved this by quoting the keys. Than also the reporting log method will show the message correct.  This looks not good and you may have an better idea.

Reply:
Done.